ansible/roles/local_storage/templates/localvolume.yml.j2

```diff
@@ -0,0 +1,22 @@
+---
+apiVersion: "local.storage.openshift.io/v1"
+kind: "LocalVolume"
+metadata:
+  name: "local-disks"
+  namespace: "openshift-local-storage" 
+spec:
+  nodeSelector: 
+    nodeSelectorTerms:
+    - matchExpressions:
+      - key: beta.kubernetes.io/instance-type
+        operator: In
+        values:
+        - "{{local_storage_instance_type}}"
```

Comment:
Can we take as input from user storage node names and the volumes that can be used in each storage node in group_vars/all.yml?
```storage_nodes:
      storage002-r640:
         - nvme0n1
         - nvme0n2
      storage003-r640:
         - nvme0n1
```

Idea would be to label the nodes provided by the user and use that label in the nodeselector here. So we take as input the names of the storage nodes and devices, then label the nodes with the tag and then use that tag in the nodeselecto here along with the devicepaths specified by user.

Reply:
I am unsure I completely follow the idea here.  Would it be easier to label all storage nodes with a label then use that? (Ex: storagenode=true) Then use the label to only place local-storage provisioner and diskmaker pods such that pvs are created only on those nodes?  If the nodes have different amount of disks I think we can just put all devices down and the provisioner will only expose the ones actually there.  (I will run a test to verify this behavior.)

Reply:
FYI I was able to create a single localvolume object that creates pvs on two nodes with different exposed lvm volumes.  

To better illustrate what I did:
Node1 device:
/dev/vg_ls/lv_tv00

Node2 devices:
/dev/vg_ls/lv_tv00
/dev/vg_ls/lv_tv01

Local volume object:
```
...
 spec:
    logLevel: Normal
    managementState: Managed
    nodeSelector:
      nodeSelectorTerms:
      - matchExpressions:
        - key: beta.kubernetes.io/instance-type
          operator: In
          values:
          - nvme
    storageClassDevices:
    - devicePaths:
      - /dev/vg_ls/lv_tv00
      - /dev/vg_ls/lv_tv01
      fsType: xfs
      storageClassName: local-sc
      volumeMode: Filesystem
```

Resulting pvs:
```
$ oc get pv 
NAME                CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS   REASON   AGE
local-pv-80f17c52   140Gi      RWO            Delete           Available           local-sc                4m5s
local-pv-b892b545   140Gi      RWO            Delete           Available           local-sc                4m5s
local-pv-f14329b1   140Gi      RWO            Delete           Available           local-sc                4m10s
$ oc describe pv | egrep "hostname|path" -i
    Term 0:        kubernetes.io/hostname in [shiftstack0059-5lf4b-worker-0-h6p5t]
    Path:  /mnt/local-storage/local-sc/dm-5
    Term 0:        kubernetes.io/hostname in [shiftstack0059-5lf4b-worker-0-h6p5t]
    Path:  /mnt/local-storage/local-sc/dm-6
    Term 0:        kubernetes.io/hostname in [shiftstack0059-5lf4b-worker-0-lpxxh]
    Path:  /mnt/local-storage/local-sc/dm-5
```

Looks like we can just label each storage node and put all paths to all devices and local storage operator will create the resulting pvs.